cli/cppcheckexecutor.cpp

```diff
@@ -408,8 +410,8 @@ int CppCheckExecutor::check_internal(CppCheck& cppcheck)
         }
     }
 
-    if (settings.xml) {
-        reportErr(ErrorMessage::getXMLFooter());
+    if (settings.xml || settings.sarif) {
+        reportErr(mReport->serialize());
```

Comment:
I suggest that `reportErr(mReport->serialize());` is called no matter what output format there is. For a CLI output format the method could return an empty string.

Reply:
Done.